Display job name in GitHub "Check" Report if it exists

jrkong opened the pull request:
Fixes #10028

Updated the table logic to check if a job name exists before defaulting to the default job number/id.

Regarding tests, how should they be handled? At the moment the `svenfuchs/minimal` repo doesn't have a job name that could be used in a test.

BanzaiMan commented:
I think the data https://github.com/travis-ci/travis-ci/issues/10028 is looking for is the build stage name. Each individual job does not have a name.

jrkong commented:
@BanzaiMan I've updated the logic to incorporate the build stage name into the job column of the chart and I've updated the test accordingly. Hopefully I targeted the right variable this time. Please let me know if there's anything else I need to fix.

BanzaiMan commented on lib/travis/addons/github_check_status/output/stage.rb:
```diff
@@ -61,10 +61,19 @@ def notes(job)
+      if stage
+        line = stage[:name] + ", " + job[:number]
+      end
+      return line
```
Looks a little wordy. Can be a little more concise without being too terse. Also, `job_name` is not exactly what this method returns. Perhaps something like:

```ruby
def job_display_text(job)
  stage ? "#{stage[:name]}, #{job[:number]}" : job[:number]
end
```

jrkong commented:
@BanzaiMan Updated the logic and changes based on your advice, thanks! I also merged the upstream changes into my branch to verify that there are no merge conflicts. Let me know if there's anything else I can fix.

BanzaiMan commented on lib/travis/addons/github_check_status/output/stage.rb:
```diff
@@ -61,10 +61,14 @@ def notes(job)
     end
 
+    def job_display_text(job)
+      stage ? "#{stage[:name]}, #{job[:number]}" : job[:number]
```
The data you can use for the job name is in `job[:config]`. You'd check if the `:name` key exists in configuration passed, and if there is, use it. I feel that this is brittle, but, for now, it is our best option.

Also, I think the name can be overridden in `travis-web` display without the presence of Build Stages, so we should follow suit.

As for the actual text, perhaps we also follow suit display the job number first, then the name.

Something like:

    job[:config].key?(:name) ? "#{job[:number]} #{job[:config][:name]}" : job[:number].to_s

BanzaiMan commented:
Also, could you rebase this, instead of merging `master`? I think the diff is a bit too busy.

jrkong commented:
Added the suggested changes and rebased the code for a cleaner diff. I'm not entirely sure how I would write a test for the feature as the test repos don't use the job name. If you tell me what I need to do to set up the required tests I'll do that as well. Let me know if there's anything else I need to change.
